src/go/rpk/pkg/cli/cmd/container/common/common.go

```diff
@@ -83,32 +82,41 @@ func ClusterDir() string {
 	return filepath.Join(home, ".rpk", "cluster")
 }
 
-func GetExistingNodes(fs afero.Fs) ([]uint, error) {
-	nodeIDs := []uint{}
-	nodeDirs, err := afero.ReadDir(fs, ClusterDir())
+func GetExistingNodes(c Client) ([]*NodeState, error) {
+	regExp := `^/rp-node-[\d]+`
```

Comment:
nice. would be nice to get a test for this at some point. 

not a blocker, but we should have a TODO

Reply:
i see a problem here, which you can partially solve by adding metadata to the network in docker. 

which is you can discover the nodes attached to the network by adding labels. so even if the directory `~/.rpk/node-1/` doesn't exist but you have a label called `node-list: 1,3,5` or whatever you know that a node directory is supposed to be there

so you can let docker network labels be a key-value database for you.

Reply:
i did this with bamboo. 

in any case. given that `rpk container ` is broken, we can add this as a quick follow up. /issue

Reply:
That sounds good. Even if we're no longer using any files/ volumes.